src/client/api/object.c

```diff
@@ -382,33 +426,41 @@ daos_oit_open(daos_handle_t coh, daos_epoch_t epoch,
 	}
 
 	if (co_global_ver < 2) {
-		oid = daos_oit_gen_id(epoch, cont_rf);
-		goto open;
+		rc = dc_obj_open_task_create(coh, daos_oit_gen_id(epoch, cont_rf),
+					     DAOS_OO_RW, oh, ev, NULL, &task);
+		if (rc)
+			return rc;
+
+		return dc_task_schedule(task, true);
+	}
+
+	D_ALLOC_PTR(ooa);
+	if (ooa == NULL) {
+		rc = -DER_NOMEM;
+		goto failed;
 	}
 
-	rc = dc_task_create(dc_cont_snap_oit_oid_get, NULL, NULL, &task);
+	rc = dc_task_create(dc_cont_snap_oit_oid_get, NULL, ev, &task);
```

Comment:
it's better to use a different task variable for obj_open and oit_open_cb. what if for example the first did not complete. technically here open is local operation, so it would complete, but still it's better to use separate tasks to make sure and guard against future changes to open.